presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/db/FunctionNamespaceDao.java

```diff
@@ -0,0 +1,240 @@
+            "    t.return_type,\n" +
+            "    t.description,\n" +
+            "    t.language,\n" +
+            "    t.`deterministic`,\n" +
+            "    t.called_on_null_input,\n" +
+            "    t.body,\n" +
+            "    t.version\n" +
+            "FROM <table_name> t\n" +
+            "JOIN (\n" +
+            "    SELECT\n" +
+            "        function_id,\n" +
+            "        MAX(version) version\n" +
+            "    FROM <table_name>\n" +
+            "    WHERE catalog_name = :catalog_name\n" +
+            "      AND schema_name = :schema_name\n" +
+            "      AND function_name = :function_name\n" +
+            "    GROUP BY\n" +
+            "        function_id\n" +
+            ") v\n" +
+            "    ON t.function_id = v.function_id\n " +
+            "    AND t.version = v.version\n" +
+            "WHERE\n" +
+            "    NOT t.deleted")
+    List<SqlInvokedFunction> getFunctions(
+            @Define("table_name") String tableName,
+            @Bind("catalog_name") String catalogName,
+            @Bind("schema_name") String schemaName,
+            @Bind("function_name") String functionName);
+
+    @SqlQuery("SELECT\n" +
+            "    catalog_name,\n" +
+            "    schema_name,\n" +
+            "    function_name,\n" +
+            "    parameters,\n" +
+            "    return_type,\n" +
+            "    description,\n" +
+            "    language,\n" +
+            "    `deterministic`,\n" +
+            "    called_on_null_input,\n" +
+            "    body,\n" +
+            "    version\n" +
+            "FROM <table_name>\n" +
+            "WHERE\n" +
+            "    function_id = :function_id\n" +
+            "    AND version = :version")
+    Optional<SqlInvokedFunction> getFunction(
+            @Define("table_name") String tableName,
+            @Bind("function_id") SqlFunctionId functionId,
+            @Bind("version") long version);
+
+    @SqlQuery("SELECT\n" +
+            "    t.catalog_name,\n" +
+            "    t.schema_name,\n" +
+            "    t.function_name,\n" +
+            "    t.parameters,\n" +
+            "    t.return_type,\n" +
+            "    t.description,\n" +
+            "    t.language,\n" +
+            "    t.`deterministic`,\n" +
+            "    t.called_on_null_input,\n" +
+            "    t.body,\n" +
+            "    t.version,\n" +
+            "    t.deleted\n" +
+            "FROM <table_name> t\n" +
+            "JOIN (\n" +
+            "    SELECT\n" +
+            "        MAX(version) version\n" +
+            "    FROM <table_name>\n" +
+            "    WHERE\n" +
+            "        function_id = :function_id\n" +
+            ") v\n" +
+            "ON\n" +
+            "    t.version = v.version\n" +
+            "WHERE\n" +
+            "    t.function_id = :function_id\n" +
+            "FOR UPDATE")
+    Optional<SqlInvokedFunctionRecord> getLatestRecord(
```

Comment:
Name it `getLatestRecordForUpdate`? People not reading the implementation would not be aware of the side effect otherwise.